ros/src/tl_detector/light_classification/tl_classifier.py

```diff
@@ -35,13 +58,28 @@ def simple_opencv_red_color_classifier(self,image):
 
         return TrafficLight.UNKNOWN
 
-    def dl_based_classifier(self,image):
+    def dl_based_classifier(self, image):
+        image_np = np.expand_dims(np.asarray(image, dtype=np.uint8), 0)
+                     
+        # Actual detection
+        (boxes, scores, classes) = self.sess.run([self.detection_boxes, self.detection_scores, self.detection_classes], feed_dict={self.image_tensor: image_np})
+
+        # Remove unnecessary dimensions
+        scores = np.squeeze(scores)
+        classes = np.squeeze(classes)
+        
+        for i in range(len(classes)):
+	    print('class =', labels_dict[classes[i]])
+	    print('score =', scores[i])
+            if (classes[i] == 2 or classes[i] == 3) and scores[i] > 0.1: # if red or yellow light with confidence more than 10%
```

Comment:
if confidence threshold is so low for this to work, usually it means that our model is not accurate enough or not trained enough yet - not enough epochs or too small dataset.

Reply:
Yeah you are right, but the point is, it's low only from far away, and the car needs to prepare and start braking early enough, that's why it should start braking at the slightest hint of red....but when you get near the light, the accuracy shoots up to 99%. But of course, if we have more data, it will detect even the far away lights with high confidence.